extensions/wikia/PotentialMemberPageExperiments/scripts/entry-point-experiment.js

```diff
@@ -3,44 +3,87 @@ require([
 	'mw',
 	'wikia.loader',
 	'wikia.mustache',
-	'wikia.tracker'
-], function ($, mw, loader, mustache, tracker) {
+	'wikia.tracker',
+	'wikia.abTest'
+], function ($, mw, loader, mustache, tracker, abTest) {
 	'use strict';
 
 	var $banner,
 		dismissCookieName = 'pmp-entry-point-dismissed',
+		experimentName = 'POTENTIAL_MEMBER_PAGE_ENTRY_POINTS',
 		track = tracker.buildTrackingFunction({
 			category: 'potential-member-experiment',
 			trackingMethod: 'analytics'
-		});
+		}),
+		experiment,
+		experiments = {
+			TOP: function () {
+				this.type = 'top';
+				this.addEntryPoint = function () {
+					$banner.insertAfter($('#WikiaPageHeader .header-container'));
+				}
+			},
+			IN_ARTICLE: function () {
+				this.type = 'in-article';
+				this.addEntryPoint = function () {
+					var $content = $('.mw-content-text'),
+						headers = $content.children('h2'),
+						$header;
+
+					// Check if there are headers in content
+					if (headers.length >= 2) {
+						$header = headers.eq(0);
+
+						// Check if first header is not first node in content
+						if ($header.prevAll('p').length) {
+							$banner.insertBefore($header);
+						} else {
+							$banner.insertBefore(headers.eq(1));
+						}
+					} else {
+						$content.append($banner);
+					}
+				}
+			}
+		};
 
 	function init() {
 		if ($.cookie(dismissCookieName)) {
 			return;
 		}
 
-		$.when(
-			loader({
-				type: loader.MULTI,
-				resources: {
-					mustache: '/extensions/wikia/PotentialMemberPageExperiments/templates/PMPEntryPoint.mustache',
-					styles: '/extensions/wikia/PotentialMemberPageExperiments/styles/entry-point-experiment.scss'
-				}
-			})
-		).done(addEntryPoint);
+		if (setExperiment()) {
+			$.when(
+				loader({
+					type: loader.MULTI,
+					resources: {
+						mustache: '/extensions/wikia/PotentialMemberPageExperiments/templates/PMPEntryPoint.mustache',
+						styles: '/extensions/wikia/PotentialMemberPageExperiments/styles/entry-point-experiment.scss'
+					}
+				})
+			).done(addEntryPoint);
+		}
+	}
+
+	function setExperiment() {
+		var group = abTest.getGroup(experimentName);
+
+		if (experiments.hasOwnProperty(group)) {
+			experiment = new experiments[group]();
+			return true;
+		}
+
+		return false;
 	}
 
 	function addEntryPoint(resources) {
```

Comment:
And I'd rename this function to `setExperiment` not to confuse this method with an experiment's method `addEntryPoint` and move the `experiment = new experiments[group]()` line here.